### ISSUE TITLE
Wireframe mockups

### ISSUE THREAD

**Issue by Beercandy** [p2prider/p2prider#15]
![IMG_0255](https://user-images.githubusercontent.com/102021884/159228598-d0fa8bc8-a707-4c26-ba07-338ac7038e8a.JPG)

Do let me know of any doubts or confusions.

**Comment by Beercandy** [comment]
The driver can set his preferred base fare as well as what his rate is per km after that. This can be found in his hamburger settings option.

**Comment by Beercandy** [comment]
User has been given a number rather than a name to prevent bias of any sort.

**Comment by Beercandy** [comment]
On the customers side when they pick to and fro the locations will be shown from north to south visually. Pickup being half that sections (above the pickup search bar)  and drop the other half (below the drop search bar)

**Comment by p2prider** [comment]
This looks SO GOOD! Fantastic work, @Beercandy and thank you so much for the effort!

**Comment by Beercandy** [comment]
Currently there is an option to send fare request to all or favourited driver. There will be another option to select a single driver when you click on it and send him a ride request. Just writing it down for later.

**Comment by ashwingattani** [comment]
@Beercandy awesome work there! few points that we can discuss:

1. should we have the language selection as an MVP? Implementing a multi-lingual app would be time-consuming and would need particular language resources to help devs.
2. As per discussions, we are not planning to have in-app payment, for now, moving that to the user screen instead of signup flow would be more feasible. Even for users who are just exploring the application.
3. We would need an SOS CTA for users/drivers at some screen while the ride is ongoing.

PS. I liked the block feature, we can explore more on that.

**Comment by ashwingattani** [comment]
We'll also probably need drivers to upload a passport photo during verification.

**Comment by p2prider** [comment]
Couple of questions/points:

1. We can remove the register/login screen, user can enter phone number and receive OTP.
    1. If they are not already registered, they can get a welcome screen which asks if they are entering as a driver or a rider
        1. If they chose driver, they get the screen for entering their vehicle and DL details and other verification info.
        2. Agree with @ashwingattani about an option to upload a selfie from the phone's camera.
    2. If they are logged in, the server returns the account type (rider/driver) and the app switches to the relevant screen based on that.
2. On the tiles shown to the rider for the available drivers nearby, what are your thoughts on fitting more information there? I am not sure if we should increase the tile size or not. The more information to add that I am thinking of is:
    1. How far away (in minutes and/or distance) the driver currently is from pickup location.
    2. Indicator for trip request (request sent and waiting for driver to accept, or request sent and driver rejected) (maybe in case of rejection we should remove the driver entry from the list with a small animation so as to not confuse the user)
    3. Indicator for whether or not the user has traveled with this driver in the past, and if so, last rating.
    4. Anything else that I am missing which could be relevant?
3. When the driver receives a request and they click on the request to expand the tile, they should also be shown:
    1. How far away (in minutes and/or distance) the pickup location is.
    2. A button to "preview the route".
        1. NOTE: Actual behavior of this button might vary depending on availability of in-app route preview or deeplink to google maps, but it will be the same button regardless.
 
Also, in future we may want to add a "settings" screen for the user to set fare preference (i.e., max rupees/km), enable automatically sending request to top 5 nearby drivers based on rating/favorite status if they match the user's fare preferences, etc.

What do you think?

**Comment by p2prider** [comment]
> User has been given a number rather than a name to prevent bias of any sort.

Sounds like a good default. What do you think about adding an option to let them set a display name, with a hint showing that they don't have to add their real name - just a nick will do to avoid biases?

**Comment by Beercandy** [comment]
> @Beercandy awesome work there! few points that we can discuss:
> 
> 1. should we have the language selection as an MVP? Implementing a multi-lingual app would be time-consuming and would need particular language resources to help devs.

Cool shall take that screen off. 


> 2. As per discussions, we are not planning to have in-app payment, for now, moving that to the user screen instead of signup flow would be more feasible. Even for users who are just exploring the application.

Ah yes. That makes sense for people just trying it out. 


> 3. We would need an SOS CTA for users/drivers at some screen while the ride is ongoing.

Sure can add that in on the trip screen. So what does the sos do? Sends a message to your preferred contact or the cops? Or does it start audio recording everything from the moment it's pressed

> 
> PS. I liked the block feature, we can explore more on that.

(Edited by @p2prider to help fix the formatting, i.e., separate replies from OP from the block quote)

**Comment by Beercandy** [comment]
> We'll also probably need drivers to upload a passport photo during verification.

Would everyone have a passport? Or would the dl suffice?

**Comment by ashwingattani** [comment]
> > We'll also probably need drivers to upload a passport photo during verification.
> 
> Would everyone have a passport? Or would the dl suffice?

not a passport but a passport size photo*, as documents DL and RC is enough I guess.

**Comment by ashwingattani** [comment]
> > @Beercandy awesome work there! few points that we can discuss:
> > 
> > 1. should we have the language selection as an MVP? Implementing a multi-lingual app would be time-consuming and would need particular language resources to help devs.
> >    Cool shall take that screen off.
> 
> > 2. As per discussions, we are not planning to have in-app payment, for now, moving that to the user screen instead of signup flow would be more feasible. Even for users who are just exploring the application.
> >    Ah yes. That makes sense for people just trying it out.
> 
> > 3. We would need an SOS CTA for users/drivers at some screen while the ride is ongoing.
> >    Sure can add that in on the trip screen. So what does the sos do? Sends a message to your preferred contact or the cops? Or does it start audio recording everything from the moment it's pressed
> 
> > PS. I liked the block feature, we can explore more on that.

We are still in discussion on how SOS would work in issue #13 , you can add the CTA for now.

**Comment by Beercandy** [comment]
> > User has been given a number rather than a name to prevent bias of any sort.
> 
> Sounds like a good default. What do you think about adding an option to let them set a display name, with a hint showing that they don't have to add their real name - just a nick will do to avoid biases?

I'll be the first one to put my name in as p3N!s. A name could have weird effects.

**Comment by ashwingattani** [comment]
> 

Agreed, I was initially suggesting the reddit way, but this is first thing that came to my mind 😄

**Comment by p2prider** [comment]
Lol, fair enough, let's stick with the current design then :+1:

**Comment by ashwingattani** [comment]
> > > User has been given a number rather than a name to prevent bias of any sort.
> > 
> > 
> > Sounds like a good default. What do you think about adding an option to let them set a display name, with a hint showing that they don't have to add their real name - just a nick will do to avoid biases?
> 
> I'll be the first one to put my name in as p3N!s. A name could have weird effects.

Agreed, I was initially suggesting the reddit way, but this is first thing that came to my mind 😄

**Comment by ashwingattani** [comment]
> Couple of questions/points:
> 
> 1. We can remove the register/login screen, user can enter phone number and receive OTP.
>    
>    1. If they are not already registered, they can get a welcome screen which asks if they are entering as a driver or a rider
>       
>       1. If they chose driver, they get the screen for entering their vehicle and DL details and other verification info.
>       2. Agree with @ashwingattani about an option to upload a selfie from the phone's camera.
>    2. If they are logged in, the server returns the account type (rider/driver) and the app switches to the relevant screen based on that.
> 2. On the tiles shown to the rider for the available drivers nearby, what are your thoughts on fitting more information there? I am not sure if we should increase the tile size or not. The more information to add that I am thinking of is:
>    
>    1. How far away (in minutes and/or distance) the driver currently is from pickup location.
>    2. Indicator for trip request (request sent and waiting for driver to accept, or request sent and driver rejected) (maybe in case of rejection we should remove the driver entry from the list with a small animation so as to not confuse the user)
>    3. Indicator for whether or not the user has traveled with this driver in the past, and if so, last rating.
>    4. Anything else that I am missing which could be relevant?
> 3. When the driver receives a request and they click on the request to expand the tile, they should also be shown:
>    
>    1. How far away (in minutes and/or distance) the pickup location is.
>    2. A button to "preview the route".
>       
>       1. NOTE: Actual behavior of this button might vary depending on availability of in-app route preview or deeplink to google maps, but it will be the same button regardless.
> 
> Also, in future we may want to add a "settings" screen for the user to set fare preference (i.e., max rupees/km), enable automatically sending request to top 5 nearby drivers based on rating/favorite status if they match the user's fare preferences, etc.
> 
> What do you think?

1. Makes sense.
2. Adding too much info would a) congest the UI b) could be irrelevant. The more things we provide the user, the less it is for them. We should keep a cap on what is a must-have for a user and what can be sidelined.
3. distance makes sense, instead of route preview we can have an overview of the destination area like AirBnB has. would be less complex in terms of technology as well.

**Comment by p2prider** [comment]
> Adding too much info would a) congest the UI b) could be irrelevant. The more things we provide the user, the less it is for them. We should keep a cap on what is a must-have for a user and what can be sidelined.

I do agree with the point of keeping things simple.

However, information should be available to the user in some way (transparency being one of the key themes of the project). Perhaps this info can be shown by expanding the tile when the user taps on it?

Regardless, we should decide piece-by-piece which piece of information would be useful to the user from a single glance and which one can be pushed back to the expanded view. In this case, I think the first three points I listed would be appreciated from a glance at least for me. I am not sure, I could be wrong. Here is my justification:

1. Distance from pickup location: if the user is in a hurry, pickup time matters as well, and they might wanna go with a slightly higher cost auto driver who is actually closer to their pickup location.
2. Trip request indicator: Since the user will be sending requests to multiple drivers at a time, expecting them to keep track of the request status for each driver by tapping back and forth would be a pretty bad user experience IMO.
3. Indicator of prior experience with the rider: This one is more subjective and I am also not fully sure how much emphasis would be useful here. I know that on Zomato, my last rating for a dish helps me a lot when picking what to order, so my line of thinking here is derived almost entirely from that experience.

> distance makes sense, instead of route preview we can have an overview of the destination area like AirBnB has. would be less complex in terms of technology as well.

Interesting. Could you give an example of what this overview would look like and how'd you calculate/source it?

**Comment by ashwingattani** [comment]
> Interesting. Could you give an example of what this overview would look like and how'd you calculate/source it?

I would look something like this, its a screenshot from airbnb. We can use google maps API to get location specific range and show it to drivers. We would already have start location when booking.

<img width="1141" alt="Screenshot 2022-03-21 at 3 54 57 PM" src="https://user-images.githubusercontent.com/9816850/159243358-909a1d94-06ee-453b-96e5-de66f648e563.png">

**Comment by p2prider** [comment]
Ah okay. I assume you mean the following:

https://user-images.githubusercontent.com/101934185/159243701-96c9cc8f-5325-4391-9b5b-9fa3d3b2426a.png

If that is the case, then that is indeed what I meant by route. Did you think I meant the written steps of the route like this:

https://user-images.githubusercontent.com/101934185/159243865-2d04d5ab-12e8-4f80-ae3f-308ab8b4eca3.png

Anyway, sorry for the confusion!

(Edit: collapsed images as obsolete)

**Comment by ashwingattani** [comment]
@p2prider no I'm not referring to the route, I understood what you meant. I am suggesting that we show the driver the relative location where the user wants to head. Ex: If the destination is billekahalli road, we would show a zoomed-out map showing billekahalli in the center and some areas around(Ranka colony/BTM/etc) with a pointer to billekahalli.

This would allow the driver to know where he is heading, but not the exact location. Drivers get to know how far relatively the destination is and the final destination is still kept intact.

**Comment by p2prider** [comment]
Ah, I see. That makes sense actually. Good idea. Can you create a new issue for this so we can iron out the details?
